src/code42cli/cmds/alert_rules.py

```diff
@@ -151,24 +149,12 @@ def handle_row(rule_id, username):
 
 def _add_user(sdk, rule_id, username):
     user_id = get_user_id(sdk, username)
-    rules = _get_rule_metadata(sdk, rule_id)
-    try:
-        if rules:
-            sdk.alerts.rules.add_user(rule_id, user_id)
-    except Py42InternalServerError:
-        _check_if_system_rule(rules)
```

Comment:
We still need a check if it's a system rule on error, as you can't add a user manually to the system-defined ones. With these changes if you try to add/remove from one of the system rules you get a very unhelpful error.

Reply:
@timabrmsn , there have been changes in this API such that we are no longer able to check for system rules.  `get_rule_by_observation_id` returns an empty list now when it's a system rule.  The only way to check if it is a system rule would be to call `get_all()`, loop through, and find the one with observer ID.  @alanag13 and I thought that was too much and opted for an error that hinted at the possibility of modifiying a system rule. Also, it used to return 500 error, now it is 404.  There is also nothing in the error message that indicates it is a system rule.

Reply:
The error should still suggest that it may be system rule, which seems pretty helpful to me. Are you not getting that?

Reply:
Yeah, the backend changed so that attempting to view or update a system rule results in a 404 now. You can still see a system rule by viewing _all_ the rules, but we felt that the approach of basically saying that the rule _either_ doesn't exist or is a system rule was simpler. That is raised in py42 now as opposed to here.

Reply:
Oh yeah! So @timabrmsn make sure you are on the latest from master on py42 when testing these changes!

Reply:
I'm on latest py42 and I still get a generic 500 server error when trying to add a user to a system rule:

```
# code42 alert-rules add-user -u test@example.com --rule-id fdd3fe20-21d7-49dd-934a-70bca338fe14
Error: Problem making request to server.
View details in /Users/tim.abramson/.code42cli/log/code42_errors.log
```

Reply:
When I do this same exact command, I get NotFound and our new custom error

Reply:
I get that if I remove a character from the rule-id, so it's actually a non-existent rule, but for a valid system rule I still am getting the 500 and generic error message printed out.

Reply:
I worked with @unparalleled-js to clear this up -- rules have two ids, rule id and observer id, and the wrong one was being used during py42 testing yesterday. This was made further confusing by the fact that `get_by_observer_id` seems to _always_ return a 200 response (the list is just empty if the id doesn't exist). We'll be making a patch release to py42 to address this, but @timabrmsn is right, this situation still raises 500 and did not switch to be 404.